WalletWasabi/Helpers/Startup/WindowsStartupChecker.cs

```diff
@@ -0,0 +1,24 @@
+using Microsoft.Win32;
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Runtime.InteropServices;
+
+namespace WalletWasabi.Helpers
+{
+	public static class WindowsStartupChecker
+	{
+		private const string KeyPath = "SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Run";
+
+		[SuppressMessage("Interoperability", "CA1416:Validate platform compatibility", Justification = "We checked the platform one layer above this.")]
```

Comment:
This is backwards. One writes API functions so that they can be re-used (ideally). Mentioning callers is problematic in this sense because if the function is to be re-used, you can't really say what the outside world does. You can just specify what the function requires to work correctly.

```suggestion
		[SuppressMessage("Interoperability", "CA1416:Validate platform compatibility", Justification = "Platform must be checked by the caller.")]
```

Reply:
Thanks, I didnt't know this.